packages/@aws-cdk/aws-ec2-alpha/lib/transit-gateway-route.ts

```diff
@@ -0,0 +1,91 @@
+import { IResource, Resource } from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+import { CfnTransitGatewayRoute } from 'aws-cdk-lib/aws-ec2';
+import { ITransitGatewayRouteTable } from './transit-gateway-route-table';
+import { ITransitGatewayAttachment } from './transit-gateway-attachment';
+
+export interface ITransitGatewayRoute extends IResource {
+  /**
+     * The destination CIDR block for this route.
+     *
+     * Destination Cidr cannot overlap for static routes but is allowed for propagated routes.
+     * When overlapping occurs, static routes take precedence over propagated routes.
+     */
+  readonly destinationCidrBlock: string;
+
+  /**
+     * The transit gateway route table this route belongs to.
+     */
+  readonly routeTable: ITransitGatewayRouteTable;
+}
+
+export interface TransitGatewayActiveRouteProps {
+  /**
+    * The transit gateway attachment to route the traffic to.
+    */
+  readonly transitGatewayAttachment: ITransitGatewayAttachment;
+
+  /**
+    * The CIDR block used for destination matches.
+    */
+  readonly destinationCidrBlock: string;
+
+  /**
+    * The transit gateway route table you want to install this route into.
+    */
+  readonly transitGatewayRouteTable: ITransitGatewayRouteTable;
+}
+
+export interface TransitGatewayBlackholeRouteProps {
+  /**
+     * The CIDR block used for destination matches.
+     */
+  readonly destinationCidrBlock: string;
+
+  /**
+     * The transit gateway route table you want to install this route into.
+     */
+  readonly transitGatewayRouteTable: ITransitGatewayRouteTable;
+}
+
+abstract class TransitGatewayRouteBase extends Resource implements ITransitGatewayRoute {
+  public abstract routeTable: ITransitGatewayRouteTable;
+  public abstract destinationCidrBlock: string;
+}
+
+export class TransitGatewayActiveRoute extends TransitGatewayRouteBase {
```

Comment:
```suggestion
export class TransitGatewayRoute extends TransitGatewayRouteBase {
```

IMO Route is more commonly used, wdyt?

Reply:
I named it `ActiveRoute` to distinguish that it is different from the `BlackholeRoute` but I think just naming it `TransitGatewayRoute` also works since the method name is called `addRoute()` and `addBlackholeRoute()`.